### PR TITLE
[GEOS-9943] Remove dependency on jdom, use standard XML APIs

### PR DESCRIPTION
**Pull request by groldan**
[JDOM](https://mvnrepository.com/artifact/org.jdom/jdom/2.0.2) latest version is from 2012.
[JDOM-2](https://mvnrepository.com/artifact/org.jdom/jdom2/2.0.6) latest version is from 2015.

Nothing they provide can't be done without standard XML APIs.

Remove direct jdom dependencies in release modules:
- [x] `gs-wms`
- [x] `gs-restconfig`
- [x] `gs-sldservice`

Out of scope:
- community module `gs-jms-geoserver`
- community module  `gs-backup-restore`
- community module `gs-taskmanager`

---

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/restconfig/src/main/java/org/geoserver/rest/catalog/AvailableResourcesConverter.java [hunk 1]
```diff
@@ -90,14 +100,34 @@ private void writeJSON(AvailableResources t, HttpOutputMessage outputMessage)
...
+            throw new IOException(e);
+        }
+        // indent and write. If we didn't indent, the above could just write to
+        // outputMessage.getBody()
+        Transformer transformer;
         try (OutputStream os = outputMessage.getBody()) {
-            outputter.output(doc, os);
+            transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            // transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "4");
```
Comment (bradh) [inline src/restconfig/src/main/java/org/geoserver/rest/catalog/AvailableResourcesConverter.java:126]: Needed?

**Review by bradh — COMMENTED** [review]
Only had a quick look, but looks OK to me.

File: src/restconfig/src/main/java/org/geoserver/rest/catalog/AvailableResourcesConverter.java (hunk 1 above)
Comment (groldan) [reply src/restconfig/src/main/java/org/geoserver/rest/catalog/AvailableResourcesConverter.java:126]: AFAIK that's how you serialize a DOM to XML? if you mean the `OutputKeys.INDENT` property then yes, in order to preserve the behavior of serializing with indentation.

Comment (bradh) [reply src/restconfig/src/main/java/org/geoserver/rest/catalog/AvailableResourcesConverter.java:126]: I meant the commented source code.

Comment (groldan) [reply src/restconfig/src/main/java/org/geoserver/rest/catalog/AvailableResourcesConverter.java]: Ah no, that's not needed :) removed and updated the pr.

**Review by mprins — APPROVED** [review]

**Comment by aaime** [comment]
Given that JDOM has not had any release recently, it's indeed time to let it go. Pity, the standard XML API is so much more verbose, almost hurts to look at.

**Pull request merged**